File: 100-Days-Series/Day 77/ContainsDuplicate.js

// Function to check if the array has duplicate values.
const getCheck=(arr)=>{
    // Bubble Sort
    for(let i=0;i<arr.length;i++){
        for(let j=0;j<arr.length-i-1;j++){
            if(arr[j]>arr[j+1]){
                let temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
        }
    }
    // Check for duplicates in the sorted array
    for(let i=0;i<arr.length-1;i++){
        if(arr[i]==arr[i+1])
            return true;
    }
    return false;
}
// Let's Start Here
const arr1=[1,2,3,1];
const arr2=[1,2,3,4];
const arr3=[1, 1, 1, 3, 3, 4, 2, 4, 2];
result1=getCheck(arr1);
result2=getCheck(arr2);
result3=getCheck(arr3);
console.log("Result 1 : "+result1);
console.log("Result 2 : "+result2);
console.log("Result 3 : "+result3);

File: 100-Days-Series/Day 77/ContainsDuplicate.java
public class ContainsDuplicate {
    // Method to check if the array has duplicate values
    public static boolean getCheck(int arr[]){
        // Bubble Sort
        for(int i=0;i<arr.length;i++){
            for(int j=0;j<arr.length-i-1;j++){
                if(arr[j]>arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                }
            }
        }
        // Check for duplicates in the sorted array
        for(int i=0;i<arr.length-1;i++){
            if(arr[i]==arr[i+1]){
                return true;
            }
        }
        return false;
    }
    // Main Method 
    public static void main(String [] args){
        // Let's Start Here
        int arr1[]={1, 2, 3, 1};
        int arr2[]={1, 2, 3, 4};
        int arr3[]={1, 1, 1, 3, 3, 4, 2, 4, 2};
        boolean result1=getCheck(arr1);
        boolean result2=getCheck(arr2);
        boolean result3=getCheck(arr3);
        System.out.println("Result 1 : "+result1);
        System.out.println("Result 1 : "+result2);
        System.out.println("Result 1 : "+result3);

    }
}
